src/Processors/Formats/Impl/TabSeparatedRowInputFormat.cpp

```diff
@@ -22,9 +24,13 @@ namespace ErrorCodes
 
 /** Check for a common error case - usage of Windows line feed.
   */
+template<bool supports_crlf>
 static void checkForCarriageReturn(ReadBuffer & in)
```

Comment:
We just should not call this check if `format_settings.tsv.crlf_end_of_line_input`

Reply:
Also we can modify the exception message and say that the user can enable new setting to read such data